scilpy/connectivity/connectivity.py

```diff
@@ -0,0 +1,197 @@
+import os
+import sys
+import threading
+
+from dipy.io.utils import is_header_compatible, get_reference_info
+from dipy.tracking.streamlinespeed import length
+from dipy.tracking.vox2track import _streamlines_in_mask
+import h5py
+import nibabel as nib
+import numpy as np
+
+from scilpy.image.labels import get_data_as_labels
+from scilpy.io.hdf5 import (assert_header_compatible_hdf5,
+                            reconstruct_streamlines_from_hdf5)
+from scilpy.tractanalysis.reproducibility_measures import \
+    compute_bundle_adjacency_voxel
+from scilpy.tractanalysis.streamlines_metrics import compute_tract_counts_map
+from scilpy.utils.metrics_tools import compute_lesion_stats
+
+d = threading.local()
+
+
+def load_node_nifti(directory, in_label, out_label, ref_img):
+    in_filename = os.path.join(directory,
+                               '{}_{}.nii.gz'.format(in_label, out_label))
+
+    if os.path.isfile(in_filename):
+        if not is_header_compatible(in_filename, ref_img):
+            raise IOError('{} do not have a compatible header'.format(
+                in_filename))
+        return nib.load(in_filename).get_fdata(dtype=np.float64)
+
+    return None
+
+
+def multi_proc_compute_connectivity_matrices_from_hdf5(args):
+    (hdf5_filename, labels_img, comb,
+     compute_volume, compute_streamline_count, compute_length,
+     similarity_directory, metrics_data, metrics_names, lesion_data,
+     include_dps, weighted, min_lesion_vol) = args
+    return compute_connectivity_matrices_from_hdf5(
+        hdf5_filename, labels_img, comb[0], comb[1],
+        compute_volume, compute_streamline_count, compute_length,
+        similarity_directory, metrics_data, metrics_names, lesion_data,
+        include_dps, weighted, min_lesion_vol)
+
+
+def compute_connectivity_matrices_from_hdf5(
+        hdf5_filename, labels_img, in_label, out_label,
+        compute_volume=True, compute_streamline_count=True,
+        compute_length=True, similarity_directory=None, metrics_data=None,
+        metrics_names=None, lesion_data=None, include_dps=False,
+        weighted=False, min_lesion_vol=0):
+    """
+    Parameters
+    ----------
+    hdf5_filename: str
+        Name of the hdf5 file containing the precomputed connections (bundles)
+    labels_img: np.ndarray
+        Data as labels
+    in_label: str
+        Name of one extremity of the bundle to analyse.
+    out_label: str
+        Name of the other extremity. Current node is {in_label}_{out_label}.
+    compute_volume: bool
+        If true, return 'volume' in the returned dictionary with the volume of
+        the bundle.
+    compute_streamline_count: bool
+        If true, return 'streamline_count' in the returned dictionary, with
+        the number of streamlines in the bundle.
+    compute_length: bool
+        If true, return 'length' in the returned dictionary, with the mean
+        length of streamlines in the bundle.
+    similarity_directory: str
+        If not None, ??
```

Comment:
?? :D

Reply:
@frheault  I wasn't sure how to explain this value. Can you help please?

Reply:
It is a directory with nifti that represents density maps for each connection using the {in_label}_{out_label}.nii.gz conventions. Typically computed from a template (must be in the same space)